pkg/packaging/packaging_test.go

```diff
@@ -136,6 +139,67 @@ func TestHelperProcess(t *testing.T) {
 
 }
 
+func Test_getBinary(t *testing.T) {
+	t.Parallel()
+
+	if runtime.GOOS == "windows" {
+		t.SkipNow()
+	}
+
+	// Set up cache directory with only a launcher binary in it
+	tmpCacheDir := t.TempDir()
+	binaryName := "launcher"
+	version := "nightly"
+	localBinaryDir := filepath.Join(tmpCacheDir, fmt.Sprintf("launcher-%s-%s", runtime.GOOS, version))
+	assert.NoError(t, os.Mkdir(localBinaryDir, 0755), "could not make temp cache directory")
+	cachedBinaryPath := filepath.Join(localBinaryDir, binaryName)
+	f, err := os.Create(cachedBinaryPath)
+	assert.NoError(t, err, "could not create binary")
+	defer f.Close()
+
+	// Set up output directory
+	tmpPkgRoot := t.TempDir()
+	binDir := "bin"
+	assert.NoError(t, os.Mkdir(filepath.Join(tmpPkgRoot, binDir), 0755), "could not make temp output directory")
+
+	p := &PackageOptions{
+		packageRoot: tmpPkgRoot,
+		binDir:      binDir,
+	}
+
+	// Verify we found the non-app bundle binary and copied it to the expected location
+	err = p.getBinary(context.TODO(), binaryName, binaryName, cachedBinaryPath)
```

Comment:
This will cause the CI system to download launcher. I'm not sure what I think. I kinda want it, but it's potentially slow and error prone.

Reply:
It doesn't -- because the cached binary path is set appropriately, we don't make the call out to `FetchBinary` and just look at the one that we set up above.

Reply:
Oh, neat! So it's passing a file system path and then it's all just doing the right thing?